src/dualscreen.tsx

```diff
@@ -72,18 +71,20 @@ export class DualScreen extends KalturaPlayer.core.BasePlugin implements IEngine
     return this._readyPromise;
   }
 
-  getUIComponents() {
-    return [
-      {
-        presets: PRESETS,
-        container: ReservedPresetAreas.BottomBarRightControls,
-        get: KalturaPlayer.ui.components.Remove,
-        replaceComponent: KalturaPlayer.ui.components.Settings.displayName
-      }
-    ];
+  private _removeSettingsComponent = () => {
+    const removeSettings =
+        {
+          presets: PRESETS,
+          container: ReservedPresetAreas.BottomBarRightControls,
+          get: KalturaPlayer.ui.components.Remove,
+          replaceComponent: KalturaPlayer.ui.components.Settings.displayName
+        }
+    this._undoRemoveSettings = this._player.ui.addComponent(removeSettings);
+    this.secondaryKalturaPlayer = this._createSecondaryPlayer();
```

Comment:
doesn't make sense to create the secondary player in _removeSettingsComponent
It is called coupling...
Better to call the create player in line 555

Reply:
I can move it to line 555 and 582 since _removeSettingsComponent is being called from the video-sync-manager as well